rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -1083,6 +1090,40 @@ rmw_take(const rmw_subscription_t * subscription, void * ros_message, bool * tak
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+rmw_take(const rmw_subscription_t * subscription, void * ros_message, bool * taken)
+{
+  return _take(subscription, ros_message, taken, nullptr);
+}
+
+rmw_gid_t *
+rmw_take_with_gid(const rmw_subscription_t * subscription, void * ros_message, bool * taken)
```

Comment:
It returns a nullptr to indicate an error and sets the error string. This is a pretty common thing across all of our functions. If we want to return an error specific error code we need to refactor many of our rmw functions. I don't want to do that here.